feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/delegation/delegation/create/chooseAmount/NewDelegationChooseAmountViewModel.kt

```diff
@@ -131,10 +151,10 @@ class NewDelegationChooseAmountViewModel(
         originFeeMixin.connectWith(
             inputSource1 = amountChooserMixin.backPressuredAmount,
             inputSource2 = selectedConvictionFlow,
-            scope = this,
-            feeConstructor = { amount, conviction ->
+            inputSource3 = chainFlow,
+            feeConstructor = { feePaymentCurrency, amount, conviction, chain ->
                 interactor.estimateFee(
-                    amount = amount.toPlanks(),
+                    amount = feePaymentCurrency.toFeePaymentAsset(chain).planksFromAmount(amount),
```

Comment:
I feel like using `FeePaymentCurrency = Native | Custon(Asset)` was not a great idea since now I can see we cannot really extract `asset` from it without passing `chain` from somewhere. I think at some point we should switch to inline class that wraps `Chain.Asset`. Wdyt?

Reply:
Mentioned issue is solved by introducing `amountInputMixin.backpressuredPlanks` since we should not rely on the fee mixin data at the first place for planks conversion